app/screens/onboarding/index.tsx

```diff
@@ -0,0 +1,160 @@
+    const translateX = useSharedValue(dimensions.width);
+
+    const currentIndex = useDerivedValue(() => Math.round(scrollX.value / width));
+
+    const moveToSlide = useCallback((slideIndexToMove: number) => {
+        slidesRef.current?.scrollTo({
+            x: (slideIndexToMove * width),
+            animated: true,
+        });
+    }, []);
+
+    const nextSlide = useCallback(() => {
+        const nextSlideIndex = currentIndex.value + 1;
+        if (slidesRef.current && currentIndex.value < LAST_SLIDE_INDEX) {
+            moveToSlide(nextSlideIndex);
+        } else if (slidesRef.current && currentIndex.value === LAST_SLIDE_INDEX) {
+            signInHandler();
+        }
+    }, [currentIndex.value, slidesRef.current, moveToSlide]);
+
+    const signInHandler = useCallback(() => {
+        // mark the onboarding as already viewed
+        storeOnboardingViewedValue();
+
+        goToScreen(Screens.SERVER, '', {animated: true, theme}, loginAnimationOptions());
+    }, []);
+
+    const renderSlide = useCallback(({item, index}: ListRenderItemInfo<OnboardingItem>) => {
+        return (
+            <SlideItem
+                item={item}
+                theme={theme}
+                scrollX={scrollX}
+                index={index}
+                key={`key-${index.toString()}`}
+                lastSlideIndex={LAST_SLIDE_INDEX}
+            />
+        );
+    }, [theme]);
+
+    const scrollHandler = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
+        scrollX.value = event.nativeEvent.contentOffset.x;
+    };
+
+    useEffect(() => {
+        const listener = {
+            componentDidAppear: () => {
+                translateX.value = 0;
+            },
+            componentDidDisappear: () => {
+                translateX.value = -dimensions.width;
+            },
+        };
+        const unsubscribe = Navigation.events().registerComponentListener(listener, Screens.ONBOARDING);
+
+        return () => unsubscribe.remove();
+    }, [dimensions]);
+
+    const transform = useAnimatedStyle(() => {
+        const duration = Platform.OS === 'android' ? 250 : 350;
+        return {
+            transform: [{translateX: withTiming(translateX.value, {duration})}],
+        };
+    }, []);
+
+    return (
+        <View
+            style={styles.onBoardingContainer}
+            testID='onboarding.screen'
+        >
+            <Background theme={theme}/>
+            <AnimatedSafeArea
+                style={[styles.scrollContainer, transform]}
+                key={'onboarding_content'}
+            >
+                <ScrollView
+                    scrollEventThrottle={16}
+                    horizontal={true}
+                    showsHorizontalScrollIndicator={false}
+                    pagingEnabled={true}
+                    bounces={false}
+                    onScroll={scrollHandler}
+                    ref={slidesRef as any}
```

Comment:
Why is this casting needed?

Reply:
was needed for the useAnimatedRef, but since the ScrollView is not longer animated, then this is no longer needed. 👍